storybookjs/storybook
# Use store revisions to ensure that stories re-render on HMR.

For #2587

Issue:

To avoid unnecessarily re-rendering stories we don't re-render if the story name+kind haven't changed.

However on HMR, this leads to not rerendering / breakage, as in this case the name/kind haven't changed, but the store's implementation itself changed.

## What I did

I added the concept of a revision to the storystore. This implies that the same name+kind can be different as the store itself has changed. I got the idea from styleguidist which I think saw something similar in when I was reading their source ;)

Moving forward, if we want dynamic stores (yes please!) we may need to generalise the concept of revision to meaning more just "incrementing on each HMR". 

## How to test

Try changing stories implementations in the kitchen sink.

Is this testable with jest or storyshots? No

Does this need a new example in the kitchen sink apps? No

Does this need an update to the documentation? No

**We should ensure this is merged into https://github.com/storybooks/storybook/pull/2241**
